hpx/parallel/algorithms/transfer.hpp

```diff
@@ -0,0 +1,155 @@
+                    boost::is_same<std::output_iterator_tag, output_iterator_category>
+            >::type is_seq;
+
+            return Algo<std::pair<InIter, OutIter>>().call(
+                    std::forward<ExPolicy>(policy), is_seq(),
+                    first, last, dest);
+        }
+
+        // forward declare segmented version
+        template <
+            template <typename IterPair> class Algo,
+            typename ExPolicy, typename InIter, typename OutIter>
+        typename util::detail::algorithm_result<
+                ExPolicy, std::pair<InIter, OutIter>
+            >::type
+        transfer_(ExPolicy && policy, InIter first, InIter last, OutIter dest,
+            std::true_type);
+
+        /// \endcond
+    }
+
+    /// Executes transfer algorithm on the elements in the range [first, last), to another range
+    /// beginning at \a dest.
+    ///
+    /// \note   Complexity: Performs exactly \a last - \a first transfer assignments.
+    ///
+    ///
+    /// \tparam Algo        The algorithm that is used to transfer the elements.
+    ///                     Should be hpx::parallel::detail::copy or
+    ///                     hpx::parallel::detail::move.
+    ///
+    /// \tparam ExPolicy    The type of the execution policy to use (deduced).
+    ///                     It describes the manner in which the execution
+    ///                     of the algorithm may be parallelized and the manner
+    ///                     in which it executes the move assignments.
+    /// \tparam InIter      The type of the source iterators used (deduced).
+    ///                     This iterator type must meet the requirements of an
+    ///                     input iterator.
+    /// \tparam OutIter     The type of the iterator representing the
+    ///                     destination range (deduced).
+    ///                     This iterator type must meet the requirements of an
+    ///                     output iterator.
+    ///
+    /// \param policy       The execution policy to use for the scheduling of
+    ///                     the iterations.
+    /// \param first        Refers to the beginning of the sequence of elements
+    ///                     the algorithm will be applied to.
+    /// \param last         Refers to the end of the sequence of elements the
+    ///                     algorithm will be applied to.
+    /// \param dest         Refers to the beginning of the destination range.
+    ///
+    /// \returns  The \a transfer algorithm returns a \a hpx::future<OutIter> if
+    ///           the execution policy is of type
+    ///           \a sequential_task_execution_policy or
+    ///           \a parallel_task_execution_policy and
+    ///           returns \a OutIter otherwise.
+    ///           The \a move algorithm returns the output iterator to the
+    ///           element in the destination range, one past the last element
+    ///           transfered.
+    ///
+    template <
```

Comment:
Please adapt this documentation to reflect the correct return type. Also, this comment should not be used for doxygen documentation generation, please change `///` to `//`.